File: app/src/main/java/com/capstone/project_niyakneyak/ui/login/etc/LoginResult.java
package com.capstone.project_niyakneyak.ui.login.etc;

import androidx.annotation.Nullable;

/**
 * Authentication result : success (user details) or error message.
 */
public class LoginResult {
    @Nullable
    private LoggedInUserView success;
    @Nullable
    private Integer error;

    public LoginResult(@Nullable Integer error) {
        this.error = error;
    }

    public LoginResult(@Nullable LoggedInUserView success) {
        this.success = success;
    }

    @Nullable
    public LoggedInUserView getSuccess() {
        return success;
    }

    @Nullable
    public Integer getError() {
        return error;
    }
}

File: app/src/main/java/com/capstone/project_niyakneyak/login/etc/LoginResult.kt
package com.capstone.project_niyakneyak.login.etc

/**
 * Authentication result : success (user details) or error message.
 */
class LoginResult {
    var success: LoggedInUserView? = null
        private set
    var error: Exception? = null
        private set

    constructor(error: Exception?) {
        this.error = error
    }

    constructor(success: LoggedInUserView?) {
        this.success = success
    }
}